internal/pkg/api/handleCheckin.go

```diff
@@ -234,27 +237,54 @@ func (ct *CheckinT) ProcessRequest(zlog zerolog.Logger, w http.ResponseWriter, r
 		}
 	}
 
-	for _, action := range actions {
-		zlog.Info().
-			Str("ackToken", ackToken).
-			Str("createdAt", action.CreatedAt).
-			Str("id", action.Id).
-			Str("type", action.Type).
-			Str("inputType", action.InputType).
-			Int64("timeout", fromPtr(action.Timeout)).
-			Msg("Action delivered to agent on checkin")
-	}
-
 	resp := CheckinResponse{
 		AckToken: &ackToken,
 		Action:   "checkin",
 		Actions:  &actions,
 	}
 
-	return ct.writeResponse(zlog, w, r, resp)
+	return ct.writeResponse(zlog, w, r, agent, resp)
 }
 
-func (ct *CheckinT) writeResponse(zlog zerolog.Logger, w http.ResponseWriter, r *http.Request, resp CheckinResponse) error {
+func (ct *CheckinT) writeResponse(zlog zerolog.Logger, w http.ResponseWriter, r *http.Request, agent *model.Agent, resp CheckinResponse) error {
+	// TODO: only process this if we have an apm tracer
```

Comment:
@joshdover What do you mean by this?

Reply:
It would be possible to avoid doing this extra processing to construct if APM is not configured in Fleet Server. It would require detecting if the `tracer` created here is `nil`: https://github.com/elastic/fleet-server/blob/6b18cfd65ddf7bdcd30fb804e6e1d7e642e11a65/internal/pkg/server/fleet.go#L346

Reply:
added the condition in the latest commits